tools/Azure.Mcp.Tools.AzureManagedLustre/tests/Azure.Mcp.Tools.AzureManagedLustre.UnitTests/FileSystem/FileSystemUpdateCommandTests.cs

```diff
@@ -0,0 +1,249 @@
+    [InlineData("--subscription sub123 --name amlfs-01 --maintenance-day Monday", false)] // Missing resource group
+    [InlineData("--subscription sub123 --resource-group rg1 --maintenance-day Monday", false)] // Missing name
+    [InlineData("--subscription sub123 --resource-group rg1 --name amlfs-01 --maintenance-day Monday", false)]
+    [InlineData("--subscription sub123 --resource-group rg1 --name amlfs-01 --maintenance-time 00:00", false)]
+    [InlineData("--subscription sub123 --resource-group rg1 --name amlfs-01 --no-squash-nid-list nid1,nid2 --squash-uid 1000", false)] // missing gid
+    public async Task ExecuteAsync_ValidatesInputCorrectly(string args, bool shouldSucceed)
+    {
+        // Arrange
+        if (shouldSucceed)
+        {
+            _svc.UpdateFileSystemAsync(
+                Arg.Is(Sub), Arg.Is(Rg), Arg.Is(Name),
+                Arg.Any<string?>(), Arg.Any<string?>(), Arg.Any<string?>(), Arg.Any<string?>(), Arg.Any<long?>(), Arg.Any<long?>(),
+                Arg.Any<string?>(), Arg.Any<RetryPolicyOptions?>())
+                .Returns(CreateLustre());
+        }
+
+        var parseResult = _commandDefinition.Parse(args.Split(' ', StringSplitOptions.RemoveEmptyEntries));
+
+        // Act
+        var response = await _command.ExecuteAsync(_context, parseResult);
+
+        // Assert
+        Assert.Equal(shouldSucceed ? HttpStatusCode.OK : HttpStatusCode.BadRequest, response.Status);
+        if (!shouldSucceed)
+        {
+            Assert.False(string.IsNullOrWhiteSpace(response.Message));
+            Assert.True(
+                response.Message.Contains("required", StringComparison.OrdinalIgnoreCase)
+                || response.Message.Contains("provide", StringComparison.OrdinalIgnoreCase)
+                || response.Message.Contains("must be", StringComparison.OrdinalIgnoreCase)
+            );
+        }
+        else
+        {
+            Assert.NotNull(response.Results);
+        }
+    }
+
+    [Fact]
+    public async Task ExecuteAsync_MaintenanceUpdate_CallsServiceAndReturnsResult()
+    {
+        var expected = CreateLustre();
+        _svc.UpdateFileSystemAsync(Sub, Rg, Name, "Monday", "01:00", null, null, null, null, null, Arg.Any<RetryPolicyOptions?>())
+            .Returns(expected);
+
+        var args = _commandDefinition.Parse([
+            "--subscription", Sub,
+            "--resource-group", Rg,
+            "--name", Name,
+            "--maintenance-day", "Monday",
+            "--maintenance-time", "01:00"
+        ]);
+
+        var response = await _command.ExecuteAsync(_context, args);
+        Assert.Equal(HttpStatusCode.OK, response.Status);
+        await _svc.Received(1).UpdateFileSystemAsync(Sub, Rg, Name, "Monday", "01:00", null, null, null, null, null, Arg.Any<RetryPolicyOptions?>());
+
+        var json = JsonSerializer.Serialize(response.Results);
+        var result = JsonSerializer.Deserialize<UpdateResultJson>(json);
```

Comment:
```suggestion
        var result = JsonSerializer.Deserialize(json, AzureManagedLustreJsonContext..Default.FileSystemUpdateResult);
```

Then remove the `UpdateResultJson` class.